src/components/NavigationTree/state.ts

```diff
@@ -90,7 +93,52 @@ export function reducer(state: NavigationTreeState = {}, action: NavigationTreeA
 
             return newState;
         }
+        case NavigationTreeActionType.ResetNode:
+            return {
+                ...state,
+                [action.payload.path]: {
+                    ...state[action.payload.path],
+                    ...getDefaultNodeState(),
+                },
+            };
         default:
             return state;
     }
 }
+
+export function selectTreeAsList(state: NavigationTreeState, rootPath: string) {
+    const list: (NavigationTreeNodeState | NavigationTreeServiceNode)[] = [];
+
+    traverseDFS(state, rootPath, (node, level) => {
+        list.push({
+            ...node,
+            level,
+        });
+
+        if (node.collapsed) {
+            return;
+        }
+
+        if (node.loading) {
```

Comment:
don't really like this bit, will try to figure out how to add service nodes in render, not here
react-list only allows single component per item though